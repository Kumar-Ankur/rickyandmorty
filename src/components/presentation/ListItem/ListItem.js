import React from "react";
import "./ListItem.scss";
import { getCreatedYear } from "../../../utils";
import * as constants from "../../../constants";

const ListItem = (props) => {
  const { item, indx } = props;
  return (
    <div key={indx} className="flex-item">
      <div className="flex-item_container">
        <img
          src={item.image}
          alt={item.name}
          className="flex-item_container-img"
        />
        <div className="flex-item_container-intro">
          <div className="flex-item_container-name">{item.name} </div>
          <div className="flex-item_container-creation">
            {`Id: ${item.id} - created ${getCreatedYear(
              item.created
            )} years ago`}{" "}
          </div>
        </div>

        <div className="flex-item_container-detail">
          <div className="flex-item_container-detail-status">
            <div className="flex-item_container-detail-title">
              {constants.STATUS_LIST}
            </div>
            <div className="flex-item_container-detail-value">
              {item.status}
            </div>
          </div>

          <div className="flex-item_container-detail-species">
            <div className="flex-item_container-detail-title">
              {constants.SPECIES_LIST}
            </div>
            <div className="flex-item_container-detail-value">
              {item.species}
            </div>
          </div>

          <div className="flex-item_container-detail-gender">
            <div className="flex-item_container-detail-title">
              {constants.GENDER_LIST}
            </div>
            <div className="flex-item_container-detail-value">
              {item.gender}
            </div>
          </div>

          <div className="flex-item_container-detail-origin">
            <div className="flex-item_container-detail-title">
              {constants.ORIGIN_LIST}
            </div>
            <div className="flex-item_container-detail-value">
              {item.origin.name}
            </div>
          </div>

          <div className="flex-item_container-detail-location">
            <div className="flex-item_container-detail-title">
              {constants.LOCATION_LIST}
            </div>
            <div className="flex-item_container-detail-value">
              {item.location.name}
            </div>
          </div>
        </div>
      </div>
    </div>
  );
};

export default ListItem;
