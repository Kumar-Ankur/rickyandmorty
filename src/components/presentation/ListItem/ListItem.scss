.flex-item {
  height: 30rem;
  width: 20rem;
  border: 1px solid #e4e4e4;
  margin-bottom: 2rem;
  background: linear-gradient(to right, #bdc3c7, #2c3e50);
  transition: ease-in-out 1s all;
  cursor: pointer;

  &:hover {
    transform: translateY(-1rem);
    box-shadow: 0 10px 20px rgba(black, 0.2);
  }

  &_container {
    margin: 0.5rem;
    transition: ease-in-out 1s all;

    &-img {
      height: 19rem;
      width: 100%;
      transition: ease-in-out 1s all;
    }

    &-intro {
      width: 100%;
      height: 4.5rem;
      position: relative;
      top: -4.65rem;
      z-index: 100;
      background: rgba(0, 0, 0, 0.7);
    }

    &-name {
      position: absolute;
      top: 0.5rem;
      left: 0.3rem;
      font-weight: bold;
      font-size: 1.3rem;
      letter-spacing: 1px;
      color: white;
      width: 90%;
      white-space: nowrap;
      overflow: hidden;
      text-overflow: ellipsis;
    }

    @media only screen and (max-width: 949px) and (min-width: 550px),
      only screen and (max-width: 549px) and (min-width: 400px),
      only screen and (max-width: 399px) and (min-width: 350px),
      only screen and (max-width: 349px) {
      &-name {
        font-size: 0.8rem;
      }
    }

    &-creation {
      position: absolute;
      color: white;
      top: 2.2rem;
      left: 0.5rem;
    }

    @media only screen and (max-width: 949px) and (min-width: 550px),
      only screen and (max-width: 549px) and (min-width: 400px),
      only screen and (max-width: 399px) and (min-width: 350px),
      only screen and (max-width: 349px) {
      &-creation {
        font-size: 0.8rem;
      }
    }

    &-detail {
      width: 100%;
      height: 10rem;
      position: relative;
      font-size: 0.8rem;
      top: -4.65rem;
      background: #d3d3d380;

      &-status {
        position: relative;
        top: 1rem;
      }

      &-species {
        position: relative;
        top: 3rem;
      }

      &-gender {
        position: relative;
        top: 5rem;
      }

      &-origin {
        position: relative;
        top: 7rem;
      }

      &-location {
        position: relative;
        top: 9rem;
      }

      &-title {
        left: 1rem;
        position: absolute;
        color: white;
      }

      @media only screen and (max-width: 949px) and (min-width: 550px),
        only screen and (max-width: 549px) and (min-width: 400px),
        only screen and (max-width: 399px) and (min-width: 350px),
        only screen and (max-width: 349px) {
        &-title {
          left: 0.1rem;
        }
      }

      @media only screen and (max-width: 949px) and (min-width: 550px),
        only screen and (max-width: 549px) and (min-width: 400px),
        only screen and (max-width: 399px) and (min-width: 350px),
        only screen and (max-width: 349px) {
        &-title {
          font-size: 0.6rem;
        }
      }

      &-value {
        right: 0.5rem;
        position: absolute;
        color: gold;
      }

      @media only screen and (max-width: 949px) and (min-width: 550px),
        only screen and (max-width: 549px) and (min-width: 400px),
        only screen and (max-width: 399px) and (min-width: 350px),
        only screen and (max-width: 349px) {
        &-value {
          display: inline-block;
          width: 50%;
          white-space: nowrap;
          overflow: hidden;
          text-overflow: ellipsis;
          right: 0.1rem;
        }
      }

      @media only screen and (max-width: 949px) and (min-width: 550px),
        only screen and (max-width: 549px) and (min-width: 400px),
        only screen and (max-width: 399px) and (min-width: 350px),
        only screen and (max-width: 349px) {
        &-value {
          font-size: 0.6rem;
          right: -10px;
        }
      }
    }
  }
}

@media only screen and (max-width: 949px) and (min-width: 550px),
  only screen and (max-width: 549px) and (min-width: 400px),
  only screen and (max-width: 399px) and (min-width: 350px),
  only screen and (max-width: 349px) {
  .flex-item {
    width: 10rem;
  }
}

@media only screen and (max-width: 949px) and (min-width: 550px),
  only screen and (max-width: 549px) and (min-width: 400px),
  only screen and (max-width: 399px) and (min-width: 350px),
  only screen and (max-width: 349px) {
  .flex-item {
    width: 9rem;
  }
}


@media only screen 
and (min-device-width : 768px) 
and (max-device-width : 1024px) {
  .flex-item {
    width: 10rem;
  }
}

@media only screen 
and (min-device-width : 1024px) 
and (max-device-width : 1365px) {
  .flex-item {
    width: 14rem;
  }
}
