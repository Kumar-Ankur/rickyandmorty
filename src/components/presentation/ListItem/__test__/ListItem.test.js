import React from 'react';
import { shallow } from 'enzyme';
import ListItem from '../ListItem';

describe('check list item component', () => {
    test('generate snapshot', () => {
        const item = {
            image: '',
            name: 'test',
            id: 1,
            created: '',
            status: 'Alive',
            species: 'Human',
            gender: 'Male',
            origin: { name: 'Earth'},
            location: { name: 'Earth'}
        }
        const wrapper = shallow(<ListItem item={ item }/>);
        expect(wrapper).toMatchSnapshot();
    })
})