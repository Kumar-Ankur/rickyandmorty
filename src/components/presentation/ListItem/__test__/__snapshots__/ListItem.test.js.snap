// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`check list item component generate snapshot 1`] = `ShallowWrapper {}`;
